internal/windowsansi/winansi_test.go

```diff
@@ -0,0 +1,69 @@
+//go:build windows
+
+package windowsansi
+
+import (
+	"os"
+	"syscall"
+	"testing"
+	"unsafe"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	enableVirtualTerminalProcessing = 0x0004
+)
+
+// getConsoleMode is a helper to retrieve the current console mode for a given handle.
+func getConsoleMode(handle syscall.Handle) (uint32, error) {
+	var mode uint32
+	ret, _, err := syscall.NewLazyDLL("kernel32.dll").NewProc("GetConsoleMode").Call(uintptr(handle), uintptr(unsafe.Pointer(&mode)))
+	if ret == 0 {
+		// In Go 1.16+, err is not nil on failure. On older versions, it might be.
+		// So we return the error from the syscall call directly.
+		return 0, err
+	}
+	return mode, nil
```

Comment:
There is something I don't get either with the code or comment


```suggestion
	ret, _, err := syscall.NewLazyDLL("kernel32.dll").NewProc("GetConsoleMode").Call(uintptr(handle), uintptr(unsafe.Pointer(&mode)))
	if ret == 0 {
		// In Go 1.16+, err is not nil on failure. On older versions, it might be.
		// So we return the error from the syscall call directly.
		return 0, err
	}
	return mode, nil
```

Reply:
The documentation says:

> “The returned error is always non-nil, constructed from the result of GetLastError. Callers must inspect the primary return value to decide whether an error occurred (according to the semantics of the specific function being called) before consulting the error.”

But since the comment was a bit confusing anyway, I’ve updated it.